*** ido2db	Fri May  4 10:37:30 2012
--- ido2db_diff	Fri May  4 10:38:57 2012
***************
*** 1,37 ****
  #!/bin/sh
! # Startup script for ndo-daemon
  #
  # chkconfig: 2345 95 05
! # description: Nagios Database Objects daemon
  
  # Source function library.
  . /etc/rc.d/init.d/functions
  
! 
! BINARY=ndo2db-3x
! DAEMON=/usr/sbin/$BINARY
! CONFIG=/etc/nagios/ndo2db.cfg
! 
! [ -f $DAEMON ] || exit 0
! 
! prog="ndo2db"
  
  start() {
      echo -n $"Starting $prog: " 
!     daemon --user nagios $DAEMON -c $CONFIG
!     RETVAL=$?
      echo
!     return $RETVAL
  }
  
  stop() {
!     if test "x`pidof $BINARY`" != x; then
          echo -n $"Stopping $prog: "
!         killproc ndo2db-3x
          echo
      fi
!     RETVAL=$?
!     return $RETVAL
  }
  
  case "$1" in
--- 1,42 ----
  #!/bin/sh
! # Startup script for ido2db-daemon
  #
  # chkconfig: 2345 95 05
! # description: Icinga Database Objects daemon
  
  # Source function library.
  . /etc/rc.d/init.d/functions
  
! prog=ido2db
! user=icinga
! prefix=/usr/local/icinga
! exec=$prefix/bin/$prog
! config=$prefix/etc/ido2db.cfg
! piddir="/var/run"
! lockdir="/var/lock/subsys"
! pidfile="$piddir/$prog.pid"
! lockfile="${lockdir}/$prog"
  
  start() {
+     [ -x $exec ] || exit 5
+     [ -f $config ] || exit 6
      echo -n $"Starting $prog: " 
!     daemon --user $user $exec -c $config
!     retval=$?
!     [ $retval -eq 0 ] && touch $lockfile
      echo
!     return $retval
  }
  
  stop() {
!     if test "x`pidof $prog`" != x; then
          echo -n $"Stopping $prog: "
!         killproc $prog
          echo
      fi
!     retval=$?
!     [ $retval -eq 0 ] && rm -f $lockfile $pidfile
!     return $retval
  }
  
  case "$1" in
***************
*** 44,57 ****
              ;;
  
          status)
!             status $BINARY
              ;;
          restart)
              stop
              start
              ;;
          condrestart)
!             if test "x`pidof $BINARY`" != x; then
                  stop
                  start
              fi
--- 49,62 ----
              ;;
  
          status)
!             status $prog
              ;;
          restart)
              stop
              start
              ;;
          condrestart)
!             if test "x`pidof $prog`" != x; then
                  stop
                  start
              fi
***************
*** 63,67 ****
  
  esac
  
! exit $RETVAL
  
--- 68,72 ----
  
  esac
  
! exit $retval
  
